### PR TITLE
Update dojo file structure and Readme

### PR DESCRIPTION
**Pull request by bertadevant**
Updated the readme to reflect that some dojos are now on swift too.  There is a new section about updating a dojo folder structure now. 

I have also renamed the two `ocp...` dojos to just be called `bowling` and `paperScissorsStone` and I have added the two `dependency-inversion` kata's to one folder that inside contains both ios and java version of the kata.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,53 +3,73 @@ dojos
...
-  |- 01012014-yourname-pairsname
+    |- yourChoosenLanguage
+        |- default
+        |- 01012014-yourname-pairsname
 ```
 
-- add the copied kata to the root `settings.gradle` as `theKata:01012014-yourname-pairsname`
+## Android katas
 
+- add the copied kata to the root `settings.gradle` as `theKata:01012014-yourname-pairsname`
```
Comment (rock3r) [inline README.md]: Can we capitalise the points? `add` -> `Add` etc (same for other sections)

File: README.md [hunk 2]
```diff
@@ -3,53 +3,73 @@ dojos
...
-```groovy
-subprojects {
-	apply from: '../default/build.gradle'
-}
-```
-should be added to keep all the implementations using the same build
-
+- create two directories inside the dojo's named `swift` and `java`
+- create a `default` directory inside whichever language you want to run the kata on
+- this `default` directory will contain any template source code and build.graddle file (if you are using a a java project)
```
Comment (rock3r) [inline README.md]: typo: _build.graddle_ --> `build.gradle`; _java_ --> _Java_

File: README.md [hunk 3]
```diff
@@ -3,53 +3,73 @@ dojos
...
 - implementations of the dojo should be at the same directory level as the default directory
- 
+
 ```
 -dojos
 |- myKata
```
Comment (rock3r) [inline README.md]: There's no default here tho? I may have not understood the instructions

File: README.md [hunk 4]
```diff
@@ -3,53 +3,73 @@ dojos
...
+    |- swift
+        |- default
+        |- yourImpl
 ```
+## Android katas
 
-- add the finished kata's to the root `settings.gradle` as `dojo:implName`
+A dojo level `build.gradle` containing should be added to keep all the implementations using the same build.
+
+```groovy
```
Comment (rock3r) [inline README.md]: You can use `gradle` to be more specific as a language

File: README.md [hunk 5]
```diff
@@ -3,53 +3,73 @@ dojos
...
+subprojects {
+	apply from: '../default/build.gradle'
+}
+```
+Add the finished kata's to the root `settings.gradle` as `dojo:implName`
 
 ```
```
Comment (rock3r) [inline README.md]: Use `gradle` plz

File: README.md [hunk 6]
```diff
@@ -3,53 +3,73 @@ dojos
...
 
-- add the finished kata's to the root `settings.gradle` as `dojo:implName`
+A dojo level `build.gradle` containing should be added to keep all the implementations using the same build.
+
+```groovy
+subprojects {
+	apply from: '../default/build.gradle'
+}
+```
+Add the finished kata's to the root `settings.gradle` as `dojo:implName`
```
Comment (rock3r) [inline README.md]: Typo: _kata's_ --> _katas_ (I know it was already there)

File: README.md [hunk 7]
```diff
@@ -3,53 +3,73 @@ dojos
...
 
 ```
 include: 'myKata:default'
 ```
 
+# Updating a dojo
 
+If the dojo only contains the `default` folder, it means it has only been used for Android programming and the structure needs to be updated. Create two new folders with  `swift` and `java`  and add everything on the current source dojo folder into java.
```
Comment (rock3r) [inline README.md]: Typo: _java_ --> _Java_

File: README.md [hunk 8]
```diff
@@ -3,53 +3,73 @@ dojos
...
 
-# Practicing a new dojo
+# Running a dojo
 
-- find the dojo's module you want to attempt this will be a module under the root dir
-- inside this copy the `default` directory and name it with the date and your name or pairing name
-- this dir you just copied should have instructions to run the dojo as well as any example code
+Find the module of the dojo you want to attempt. This will be a module under the root dir. Inside you will either find the  `default`  directory or a `java` and `swift` directory.
+
+If you only find the  `default` directory, follow [these instructions](https://github.com/novoda/dojos#Updating-a-dojo). Once the folder structure has been updated properly you can continue, make a copy of the default project of whichever language you want to run the kata with.
```
Comment (rock3r) [inline README.md]: > `[these instructions](https://github.com/novoda/dojos#Updating-a-dojo)`

Tip: if it's on the same page you only need to put the anchor: `[these instructions](#Updating-a-dojo)`

File: README.md [hunk 9]
```diff
@@ -3,53 +3,73 @@ dojos
...
+	apply from: '../default/build.gradle'
+}
+```
+Add the finished kata's to the root `settings.gradle` as `dojo:implName`
 
 ```
 include: 'myKata:default'
 ```
 
+# Updating a dojo
```
Comment (lgvalle) [inline README.md]: Shall we call those "legacy"?  `Updating a legacy dojo` ?

**Comment by lgvalle** [comment]
Reviewing this I've noticed there's a `default` directory at the root of the project. It looks like every dojo it's in its own directory but there's one that's using "root" as the home directory.

File: README.md (hunk 3 above)
Comment (bertadevant) [reply README.md]: What did you not understand?

File: README.md [hunk 10]
```diff
@@ -3,53 +3,73 @@ dojos
...
-- a dojo level `build.gradle` containing 
-
-```groovy
-subprojects {
-	apply from: '../default/build.gradle'
-}
-```
-should be added to keep all the implementations using the same build
-
+- create two directories inside the dojo's named `swift` and `java`
```
Comment (tasomaniac) [inline README.md]: What about dojos that are not language dependant at all? Actually we have tons of these. They also do not come with initial source code.
So should we say something like this?

> If the dojo targets specific programming languages like `swift` or `java`, create a directory with the name of the language.

File: README.md (hunk 7 above)
Comment (tasomaniac) [inline README.md]: Actually, the majority of the dojos in the repository are not targeted for Android programming. Or they can really only be supported by Android sometimes. E.g kata around a specific library only available in Android.

We should maybe reword this to talk about when we really need to do this. Probably the case would be that the initial dojo was prepared and targeted only to java, then the 2nd language is being added.

**Comment by tasomaniac** [comment]
I feel like the document is too specific/strict with these changes now. README was not talking about any programming languages at all. It was just defining structure. Programming languages should be mentioned just to give examples, imho. 

And also imagine these scenarios:
- multi-platform projects
- javascript projects
- language independent projects

File: README.md (hunk 7 above)
Comment (bertadevant) [reply README.md]: Yeah wording this was pretty difficult because I do not know all the possible cases and it should be interesting to note possible new ones.  

Answering your comment here I think you are right, I will re write it without being language specific and taking into account that some kata's are for specific language and some have no language specification.

Comment (tasomaniac) [reply README.md]: Thanks @bertadevant 👍 
My intention was not to block anything. Feel free to shape it however you like. Can also be merged to do a follow-up later. Your call.

File: README.md [hunk 11]
```diff
@@ -3,53 +3,69 @@ dojos
...
-- a dojo level `build.gradle` containing 
+- Create a new directory inside root with the dojo's name
+- If the dojo targets specific programming languages like `Swift` or `Java`, create a directory with the name of the language. If the kata does not have a specific programming language you can place the code on root folder.
+- create a `default` directory inside whichever language you want to run the kata on
+- This `default` directory will contain any template source code and `gradle` file (if you are using a a Java project)
+- Implementations of the dojo should be at the same directory level as the default directory
+```
+-dojos
+|- myKata
+    |- oneLanguage
```
Comment (bertadevant) [inline README.md]: @tasomaniac I think this is better now

File: README.md [hunk 12]
```diff
@@ -3,53 +3,69 @@ dojos
...
-- add the finished kata's to the root `settings.gradle` as `dojo:implName`
-
+Add the finished katas to the root `gradle` as `dojo:implName`
 ```
 include: 'myKata:default'
 ```
 
+# Updating a legacy dojo
 
+If the dojo only contains the `default` folder, it is possible that it has only been used for Android programming before and the structure needs to be updated. To do so create a `Java` folder and place all the code currently on root inside of the new folder (including the `default` folder). You can then create a new folder on the kata root with the new language you want to run the kata on if necessary.
```
Comment (bertadevant) [inline README.md]: I also tried to make this non language specific.
